ModulePresentationsForCAP/gap/ModulePresentationsForCAP.gi

```diff
@@ -2146,3 +2172,296 @@ end );
     end, 1000 );
 
 end );
+
+##
+## Random methods
+##
+
+##
+## Interpretation of n:
+## The number of rows and colms of the matrix of the object is less or equal to n.
+InstallGlobalFunction( ADD_RANDOM_OBJECT_LEFT,
+
+  function( category )
+
+    AddRandomObject( category,
+
+      function( C, n )
+        local nr_rows, nr_cols, mat, homalg_ring;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        if n = 0 then
+          
+          return ZeroObject( category );
+
+        fi;
+
+        nr_rows := Random( [ 1 .. n ] );
+
+        nr_cols := Random( [ 1 .. n ] );
+
+        mat := homalg_ring!.random_matrix_func( nr_rows, nr_cols );
+
+        return AsLeftPresentation( mat );
+
+      end );
+
+end );
+
+##
+InstallGlobalFunction( ADD_RANDOM_OBJECT_RIGHT,
+
+  function( category )
+
+    AddRandomObject( category,
+        function( category, n )
+        local homalg_ring, left_presentations, M;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        left_presentations := LeftPresentations( homalg_ring );
+
+        M := RandomObject( left_presentations, n );
+
+        return AsRightPresentation( Involution( UnderlyingMatrix( M ) ) );
+
+      end );
+
+end );
+
+##
+## Interpretation of n:
+## The number of relations of the range of the created random morphism is n.
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_SOURCE_LEFT,
+    function( category )
+      local homalg_ring;
+
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedSource( category,
+        function( M, n )
+          local m, y, syz, x, u, U, e;
+          
+          if n = 0 then
+            
+            return UniversalMorphismIntoZeroObject( M );
+
+          fi;
+
+          m := UnderlyingMatrix( M );
+
+          y := homalg_ring!.random_matrix_func( NrRows( m ), n );
+
+          syz := SyzygiesOfColumns( UnionOfColumns( m, y ) );
+
+          if NrCols( syz ) > 0 then
+
+            e := homalg_ring!.random_element_func( );
+
+            e := Random( [ e, e, e, One( homalg_ring ) ] );
+
+            syz := e * CertainColumns( syz, [ 1 .. Random( [ 1 .. NrCols( syz ) ]  ) ] );
+
+          fi;
+
+          x := CertainRows( syz, [ 1 .. NrCols( m ) ] );
+
+          u := CertainRows( syz, [ NrCols( m ) + 1 .. NrRows( syz ) ] );
+
+          U := AsLeftPresentation( u );
+
+          return PresentationMorphism( M, x, U );
+
+    end );
+
+end );
+
+##
+## Interpretation of n:
+## The number of generators of the source of the created random morphism is n.
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_RANGE_LEFT,
+    function( category )
+      local homalg_ring;
+
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedRange( category,
+        function( U, n )
+          local u, x, syz, m, M, e;
+          
+          if n = 0 then
```

Comment:
In this case, the special case is justified.

Reply:
Yes, since n is the number of generators of the module.